mobile/android/android-emulator-services.ts

```diff
@@ -33,13 +38,36 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 	}
 
 	public checkDependencies(): IFuture<void> {
+		return (() => {
+			this.checkAndroidSDKConfiguration().wait();
+			if(options.geny) {
+				this.checkGenymotionConfiguration().wait();
+			}
+		}).future<void>()();
+	}
+
+	private checkAndroidSDKConfiguration(): IFuture<void> {
 		var future = new Future<void>();
 		var proc = this.$childProcess.spawn('emulator', ['-help']);
-		proc.on("error", (args:any) => {
+		proc.on("error", (args: any) => {
 			this.$errors.fail("The Android SDK is not configured properly. " +
 				"Verify that you have installed the Android SDK and that you have added its `platform-tools` and `tools` directories to your PATH environment variable.");
 		});
-		proc.on("exit", (code:number) => {
+		proc.on("exit", (code: number) => {
```

Comment:
Again use `this.$childProcess.spawnFromEvent('emulator', ['-help'], "exit", {throwError: false})`